File: app/src/main/java/com/codingstuff/loginandsignup/Activity/ResetPasswordActivity.java
package com.codingstuff.loginandsignup.Activity;

import android.app.ProgressDialog;
import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.codingstuff.loginandsignup.databinding.ActivityResetPasswordBinding;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;

public class ResetPasswordActivity extends AppCompatActivity {
    ActivityResetPasswordBinding binding;
    ProgressDialog dialog;
    FirebaseAuth auth;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityResetPasswordBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        auth = FirebaseAuth.getInstance();

        binding.btnResetET.setOnClickListener(view -> {
            forgotPassword();
        });
    }
    private Boolean validateEmail() {
        String val = binding.email.getText().toString();
        String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";

        if (val.isEmpty()) {
            binding.email.setError("Kolom tidak boleh kosong !");
            return false;
        } else if (!val.matches(emailPattern)) {
            binding.email.setError("Cek format email!");
            return false;
        } else {
            binding.email.setError(null);
            return true;
        }
    }

    private void forgotPassword() {

        if (!validateEmail())
        {
            return;
        }
        

        auth.sendPasswordResetEmail(binding.email.getText().toString()).addOnCompleteListener(new OnCompleteListener<Void>() {
            @Override
            public void onComplete(@NonNull Task<Void> task) {

                if (task.isSuccessful())
                {
                    startActivity(new Intent(ResetPasswordActivity.this, SignInActivity.class));
                    finish();
                    Toast.makeText(ResetPasswordActivity.this, "Cek Email!", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(ResetPasswordActivity.this, "Masukkan email yang benar!", Toast.LENGTH_SHORT).show();
                }
            }
        }).addOnFailureListener(new OnFailureListener() {
            @Override
            public void onFailure(@NonNull Exception e) {
                Toast.makeText(ResetPasswordActivity.this, e.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });
    }

}

File: app/src/main/java/com/codingstuff/loginandsignup/Activity/ResetPasswordActivity.kt
package com.codingstuff.loginandsignup.Activity

import android.content.Intent
import android.os.Bundle
import android.os.PersistableBundle
import android.util.Patterns
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.codingstuff.loginandsignup.R
import com.codingstuff.loginandsignup.databinding.ActivityResetPasswordBinding
import com.codingstuff.loginandsignup.databinding.ActivitySignInBinding
import com.google.firebase.auth.FirebaseAuth

class ResetPasswordActivity : AppCompatActivity() {
    private lateinit var binding: ActivityResetPasswordBinding
    private lateinit var etEmailReset: EditText
    private lateinit var btnResetPassword: Button

    private lateinit var auth: FirebaseAuth


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityResetPasswordBinding.inflate(layoutInflater)
        setContentView(binding.root)

        etEmailReset = findViewById(R.id.resetemailEt)
        btnResetPassword = findViewById(R.id.btnResetET)
        auth = FirebaseAuth.getInstance()

        binding.textRest.setOnClickListener {
            val intent = Intent(this, SignInActivity::class.java)
            startActivity(intent)
        }
        btnResetPassword.setOnClickListener{
            compareEmail(etEmailReset)
        }

    }
    //Diluar OnCreate
    private fun compareEmail(email: EditText){
        if (email.text.toString().isEmpty()){
            Toast.makeText(this, "Kolom tidak boleh kosong", Toast.LENGTH_SHORT).show()
            return
        }
        if (!Patterns.EMAIL_ADDRESS.matcher(email.text.toString()).matches()){
            Toast.makeText(this, "Cek format email!", Toast.LENGTH_SHORT).show()
            return
        }
        auth.sendPasswordResetEmail(email.text.toString()).addOnCompleteListener {task ->
            if (task.isSuccessful){
                Toast.makeText(this, "Cek Email!", Toast.LENGTH_SHORT).show()
            }
        }


    }
}